pkg/controller/signup_test.go

```diff
@@ -45,17 +47,60 @@ func (s *TestSignupSuite) TestSignupPostHandler() {
 	signupCtrl := controller.NewSignup(s.Config, svc)
 	handler := gin.HandlerFunc(signupCtrl.PostHandler)
 
-	s.Run("signup", func() {
+	s.Run("signup created", func() {
 		// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
 		rr := httptest.NewRecorder()
 		ctx, _ := gin.CreateTestContext(rr)
 		ctx.Request = req
 
+		signup := &crtapi.UserSignup{
+			TypeMeta: v1.TypeMeta{},
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "john",
+				Namespace: "namespace-foo",
+			},
+			Spec: crtapi.UserSignupSpec{
+				UserID:            "abc123",
+				Username:          "bill",
+				CompliantUsername: "bill",
+			},
+			Status: crtapi.UserSignupStatus{
+				Conditions: []crtapi.Condition{
+					{
+						Type:    crtapi.UserSignupComplete,
+						Status:  apiv1.ConditionFalse,
+						Reason:  "test_reason",
+						Message: "test_message",
+					},
+				},
+			},
+		}
+
+		svc.MockCreateUserSignup = func(username, userID string) (*crtapi.UserSignup, error) {
```

Comment:
Check if you got the expected UserID here.

Reply:
Addressed in Addressed in https://github.com/codeready-toolchain/registration-service/pull/49